source/indexes.txt

```diff
@@ -0,0 +1,258 @@
+.. _kotlin-sync-indexes:
+
+=============================
+Optimize Queries with Indexes
+=============================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :description: Learn how to use indexes with the MongoDB {+driver-short+} driver.
+   :keywords: query, optimization, efficiency, usage example, code example
+
+.. TODO
+.. .. toctree::
+..    :titlesonly:
+..    :maxdepth: 1
+
+..    /work-with-indexes
+
+Overview
+--------
+
+On this page, you can see copyable code examples that show how to work with common
+types of indexes that you can use with the {+driver-short+}.
+
+.. TODO
+.. .. tip::
+
+..   To learn more about working with indexes, see the :ref:`kotlin-sync-work-with-indexes`
+..   guide. To learn more about any of the indexes shown on this page, see the link
+..   provided in each section.
+
+To use an example from this page, copy the code example into the
+:ref:`sample application <kotlin-sync-index-sample>` or your own application.
+Be sure to replace all placeholders in the code examples, such as ``<connection string URI>``, with
+the relevant values for your MongoDB deployment.
+
+.. _kotlin-sync-index-sample:
+
+.. include:: /includes/usage-examples/sample-app-intro.rst
+
+.. literalinclude:: /includes/usage-examples/sample-app.kt
+   :language: kotlin
+   :copyable:
+   :linenos:
+   :emphasize-lines: 20-22
+
+Single Field Index
+------------------
+
+The following example creates an ascending index on the specified field:
+
+.. literalinclude:: /includes/usage-examples/index-code-examples.kt
+   :start-after: start-single-field
+   :end-before: end-single-field
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+.. TODO: To learn more about single field indexes, see the
+.. :ref:`kotlin-sync-single-field-index` guide.
+
+Compound Index
+--------------
+
+The following example creates a compound index on the specified fields:
+
+.. literalinclude:: /includes/usage-examples/index-code-examples.kt
+   :start-after: start-compound
+   :end-before: end-compound
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+.. TODO: To learn more about compound indexes, see the :ref:`kotlin-sync-compound-index`
+.. guide.
+
+Multikey Index
+--------------
+
+The following example creates a multikey index on the specified field that contains
+array values:
+
+.. literalinclude:: /includes/usage-examples/index-code-examples.kt
+   :start-after: start-multikey
+   :end-before: end-multikey
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+.. TODO To learn more about multikey indexes, see the :ref:`kotlin-sync-multikey-index`
+.. guide.
+
+Atlas Search Index
+------------------
+
```

Comment:
S: add a quick intro sentence
"The following sections describe how to manage Atlas Search indexes.."